modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java

```diff
@@ -367,67 +366,17 @@ else if (clientExtension.type.equals("themeCSS")) {
 						project, languageOAuthClientExtension.id);
 				});
 
-			TaskProvider<DefaultTask> generateLangBatchTaskProvider =
-				GradleUtil.addTaskProvider(
-					project, "generateLangBatch", DefaultTask.class);
-
-			generateLangBatchTaskProvider.configure(
-				generateLangBatchTask -> {
-					File projectDir = project.getProjectDir();
-
-					Path outputDir = Paths.get(
-						projectDir.getPath(), "build", "generateLangBatch");
-
-					Path outputPath = outputDir.resolve("batch.json");
-
-					TaskOutputs taskOutputs =
-						generateLangBatchTask.getOutputs();
-
-					taskOutputs.file(project.provider(() -> outputPath));
-
-					generateLangBatchTask.doFirst(
-						new Action<Task>() {
-
-							@Override
-							public void execute(Task task) {
-								String jsonString = "{}";
-
-								try {
-									if (!Files.exists(outputDir)) {
-										outputDir.toFile(
-										).mkdirs();
-									}
-
-									if (!Files.exists(outputPath)) {
-										Files.createFile(outputPath);
-									}
-
-									Files.write(
-										outputPath,
-										jsonString.getBytes(
-											StandardCharsets.UTF_8));
-								}
-								catch (IOException ioException) {
-									throw new GradleException(
-										"Could not write batch file",
-										ioException);
-								}
-							}
-
-						});
-				});
+			generateLanguageBatchConfigTaskProvider.configure(
+				GenerateLanguageBatchConfigTask::
+					convertPropertiesFilesToBatchFiles);
```

Comment:
I don't think this line is needed - the @TaskAnnotation takes care of running that method.

Reply:
So I just have to create as I do in the line 150 and it will automatically run?

Reply:
Correct. 

This is actually not just unnecessary, but incorrect. The `configure` method configures the task _before_ the action runs, but here you are passing in the action method as a method to be run during the configuration phase.

Reply:
Got it, thanks for the explanation